tests/plugins/stylescombo/stylescombo.js

```diff
@@ -60,6 +60,40 @@
 				assert.areSame( '<h2 style="font-style:italic">A</h2><p>X<big>Y</big>Z</p>',
 					bot.editor.dataProcessor.toHtml( '<h2 style="font-style:italic">A</h2><p>X<big>Y</big>Z</p>' ) );
 			} );
+		},
+
+		// #646
+		'test for applying style without selection': function() {
+			bender.editorBot.create( {
+				name: 'style_error',
+				config: {
+					removePlugins: 'format,font'
+				}
+			}, function( bot ) {
+				var editor = bot.editor;
+				var selection = editor.getSelection();
+
+				var stub = sinon.stub( CKEDITOR.dom.selection.prototype, 'getNative', function() {
```

Comment:
I add short comment in code before stub to have this info in case of future cases.
Generally, when combo was opened, there was execute operation which operate on selection. During this process selection was reset. So in my test I wasn't able to simulate `None` selection and always receive `Caret` selection. This behaviour affect on test and doesn't allow on proper error reproduction.
To be precise I wasn't able to get `elementPath = null` from here: https://github.com/ckeditor/ckeditor-dev/blob/master/plugins/stylescombo/plugin.js#L145
because of way how this work in those places:
https://github.com/ckeditor/ckeditor-dev/blob/master/core/editor.js#L860-L870
here is `this.getSelection()` which generally get selection based on native one here:
https://github.com/ckeditor/ckeditor-dev/blob/master/core/selection.js#L1342
And that's why I stub `getNative()` to reset native selection and always get `None` type.